### ISSUE TITLE
pip3 install pyuv failing on Windows 7

### ISSUE THREAD

**Issue by brcolow** [saghul/pyuv#219]
Full log:

````
C:\Users\brcolow> pip3 install git+https://github.com/saghul/pyuv
Collecting git+https://github.com/saghul/pyuv
  Cloning https://github.com/saghul/pyuv to c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build
Installing collected packages: pyuv
  Running setup.py install for pyuv ... error
    Complete output from command c:\python35\python.exe -u -c "import setuptools, tokenize;__file__='C:\\Users\\brcolow\\AppData\\Local\\Temp\\pip-19q8nkgx-build\\setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record C:\Users\brcolow\AppData\Local\Temp\pip-q2hv713m-record\install-record.txt --single-version-externally-managed --compile:
    running install
    running build
    running build_py
    creating build
    creating build\lib.win-amd64-3.5
    creating build\lib.win-amd64-3.5\pyuv
    copying pyuv\_version.py -> build\lib.win-amd64-3.5\pyuv
    copying pyuv\__init__.py -> build\lib.win-amd64-3.5\pyuv
    running build_ext
    Downloading libuv...
    Building libuv...
    building 'pyuv._cpyuv' extension
    creating build\temp.win-amd64-3.5
    creating build\temp.win-amd64-3.5\Release
    creating build\temp.win-amd64-3.5\Release\src
    C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\amd64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -DWIN32=1 -Ic:\python35\include -Ic:\python35\include -Ideps\libuv\include "-IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE" "-IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\ATLMFC\INCLUDE" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.10240.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\include\um" "-IC:\Program Files (x86)\Windows Kits\8.1\include\\shared" "-IC:\Program Files (x86)\Windows Kits\8.1\include\\um" "-IC:\Program Files (x86)\Windows Kits\8.1\include\\winrt" /Tcsrc/pyuv.c /Fobuild\temp.win-amd64-3.5\Release\src/pyuv.obj
    pyuv.c
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\loop.c(114): warning C4311: 'type cast': pointer truncation from 'uv_work_t *' to 'long'
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\loop.c(136): warning C4311: 'type cast': pointer truncation from 'uv_work_t *' to 'long'
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\async.c(10): warning C4311: 'type cast': pointer truncation from 'uv_async_t *' to 'long'
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\timer.c(10): warning C4311: 'type cast': pointer truncation from 'uv_timer_t *' to 'long'
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\prepare.c(10): warning C4311: 'type cast': pointer truncation from 'uv_prepare_t *' to 'long'
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\idle.c(10): warning C4311: 'type cast': pointer truncation from 'uv_idle_t *' to 'long'
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\check.c(10): warning C4311: 'type cast': pointer truncation from 'uv_check_t *' to 'long'
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\signal.c(11): warning C4311: 'type cast': pointer truncation from 'uv_signal_t *' to 'long'
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\stream.c(28): warning C4311: 'type cast': pointer truncation from 'uv_shutdown_t *' to 'long'
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\stream.c(113): warning C4311: 'type cast': pointer truncation from 'uv_write_t *' to 'long'
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\stream.c(269): warning C4244: 'function': conversion from 'Py_ssize_t' to 'unsigned int', possible loss of data
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\stream.c(312): warning C4244: 'function': conversion from 'Py_ssize_t' to 'unsigned int', possible loss of data
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\stream.c(377): warning C4244: '=': conversion from 'Py_ssize_t' to 'int', possible loss of data
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\stream.c(387): warning C4244: '=': conversion from 'Py_ssize_t' to 'ULONG', possible loss of data
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\stream.c(399): warning C4244: 'function': conversion from 'Py_ssize_t' to 'unsigned int', possible loss of data
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\stream.c(401): warning C4244: 'function': conversion from 'Py_ssize_t' to 'unsigned int', possible loss of data
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\stream.c(475): warning C4311: 'type cast': pointer truncation from 'uv_os_fd_t' to 'long'
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\pipe.c(10): warning C4311: 'type cast': pointer truncation from 'uv_pipe_t *' to 'long'
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\pipe.c(42): warning C4311: 'type cast': pointer truncation from 'uv_pipe_t *' to 'long'
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\tcp.c(11): warning C4311: 'type cast': pointer truncation from 'uv_tcp_t *' to 'long'
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\tcp.c(43): warning C4311: 'type cast': pointer truncation from 'uv_tcp_t *' to 'long'
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\udp.c(22): warning C4311: 'type cast': pointer truncation from 'uv_udp_t *' to 'long'
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\udp.c(81): warning C4311: 'type cast': pointer truncation from 'uv_udp_send_t *' to 'long'
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\udp.c(82): warning C4311: 'type cast': pointer truncation from 'uv_udp_t *' to 'long'
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\udp.c(230): warning C4244: 'function': conversion from 'Py_ssize_t' to 'unsigned int', possible loss of data
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\udp.c(270): warning C4244: 'function': conversion from 'Py_ssize_t' to 'unsigned int', possible loss of data
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\udp.c(329): warning C4244: '=': conversion from 'Py_ssize_t' to 'int', possible loss of data
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\udp.c(339): warning C4244: '=': conversion from 'Py_ssize_t' to 'ULONG', possible loss of data
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\udp.c(345): warning C4244: 'function': conversion from 'Py_ssize_t' to 'unsigned int', possible loss of data
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\udp.c(611): warning C4311: 'type cast': pointer truncation from 'uv_os_fd_t' to 'long'
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\poll.c(11): warning C4311: 'type cast': pointer truncation from 'uv_poll_t *' to 'long'
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\poll.c(113): warning C4311: 'type cast': pointer truncation from 'uv_os_fd_t' to 'long'
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\fs.c(65): warning C4311: 'type cast': pointer truncation from 'uv_fs_t *' to 'long'
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\fs.c(1058): warning C4244: 'function': conversion from 'Py_ssize_t' to 'unsigned int', possible loss of data
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\fs.c(1582): warning C4311: 'type cast': pointer truncation from 'uv_fs_event_t *' to 'long'
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\fs.c(1836): warning C4311: 'type cast': pointer truncation from 'uv_fs_poll_t *' to 'long'
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\process.c(185): warning C4311: 'type cast': pointer truncation from 'uv_process_t *' to 'long'
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\process.c(187): warning C4244: 'function': conversion from 'int64_t' to 'long', possible loss of data
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\dns.c(57): warning C4311: 'type cast': pointer truncation from 'uv_getaddrinfo_t *' to 'long'
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\dns.c(113): warning C4311: 'type cast': pointer truncation from 'uv_getnameinfo_t *' to 'long'
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\util.c(220): warning C4244: 'function': conversion from 'uint64_t' to 'long', possible loss of data
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\util.c(221): warning C4244: 'function': conversion from 'uint64_t' to 'long', possible loss of data
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\util.c(222): warning C4244: 'function': conversion from 'uint64_t' to 'long', possible loss of data
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\util.c(223): warning C4244: 'function': conversion from 'uint64_t' to 'long', possible loss of data
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\util.c(224): warning C4244: 'function': conversion from 'uint64_t' to 'long', possible loss of data
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\util.c(225): warning C4244: 'function': conversion from 'uint64_t' to 'long', possible loss of data
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\util.c(226): warning C4244: 'function': conversion from 'uint64_t' to 'long', possible loss of data
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\util.c(227): warning C4244: 'function': conversion from 'uint64_t' to 'long', possible loss of data
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\util.c(228): warning C4244: 'function': conversion from 'uint64_t' to 'long', possible loss of data
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\util.c(229): warning C4244: 'function': conversion from 'uint64_t' to 'long', possible loss of data
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\util.c(230): warning C4244: 'function': conversion from 'uint64_t' to 'long', possible loss of data
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\util.c(231): warning C4244: 'function': conversion from 'uint64_t' to 'long', possible loss of data
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\util.c(232): warning C4244: 'function': conversion from 'uint64_t' to 'long', possible loss of data
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\util.c(233): warning C4244: 'function': conversion from 'uint64_t' to 'long', possible loss of data
    c:\users\brcolow\appdata\local\temp\pip-19q8nkgx-build\src\util.c(313): warning C4311: 'type cast': pointer truncation from 'uv_poll_t *' to 'long'
    C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\amd64\link.exe /nologo /INCREMENTAL:NO /LTCG /DLL /MANIFEST:EMBED,ID=2 /MANIFESTUAC:NO /LIBPATH:c:\python35\libs /LIBPATH:c:\python35\PCbuild\amd64 "/LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\LIB\amd64" "/LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\ATLMFC\LIB\amd64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.10240.0\ucrt\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\lib\um\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\8.1\lib\winv6.3\um\x64" advapi32.lib iphlpapi.lib psapi.lib shell32.lib user32.lib userenv.lib ws2_32.lib /EXPORT:PyInit__cpyuv build\temp.win-amd64-3.5\Release\src/pyuv.obj deps\libuv\Release\lib\libuv.lib /OUT:build\lib.win-amd64-3.5\pyuv\_cpyuv.cp35-win_amd64.pyd /IMPLIB:build\temp.win-amd64-3.5\Release\src\_cpyuv.cp35-win_amd64.lib /NODEFAULTLIB:libcmt /LTCG
    LINK : fatal error LNK1181: cannot open input file 'deps\libuv\Release\lib\libuv.lib'
    error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\BIN\\amd64\\link.exe' failed with exit status 1181
    ----------------------------------------
Command "c:\python35\python.exe -u -c "import setuptools, tokenize;__file__='C:\\Users\\brcolow\\AppData\\Local\\Temp\\pip-build-b1pdk4si\\pyuv\\setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record C:\Users\brcolow\AppData\Local\Temp\pip-b55oikb3-record\install-record.txt --single-version-externally-managed --compile" failed with error code 1 in C:\Users\brcolow\AppData\Local\Temp\pip-build-b1pdk4si\pyuv\
````

System:
Windows 7 Ultimate
Service Pack 1
pip2:
pip 8.1.1 from c:\python27\lib\site-packages (python 2.7)
pip3:
pip 8.1.1 from c:\program files\python\python35\lib\site-packages (python 3.5)'

Thanks! We are trying to use pyuv for Neovim's [python-client](https://github.com/neovim/python-client) on Windows, because stdio [seems to be broken](http://stackoverflow.com/questions/31510190/aysncio-cannot-read-stdin-on-windows) in asyncio on Windows.

**Comment by saghul** [comment]
I believe this is a problem with Python 3.5 + Visual Studio 2015. Some adjustments are needed here: https://github.com/saghul/pyuv/blob/v1.x/setup_libuv.py#L65

I'll take a look as soon as I can, but I'll be out next week. If you figure it out in the mean time, a PR would be more thank welcome :-)

**Comment by brcolow** [comment]
Thanks for the pointers. Unfortunately, after trying a lot of tweaks, I kept running into linker errors so I think it is a bit outside my wheelhouse.

**Comment by brcolow** [comment]
Updated the log to reflect attempting to install the latest bleeding edge version directly using:

`pip3 install git+https://github.com/saghul/pyuv`

incase anything changed.

**Comment by saghul** [comment]
@brcolow I'll fix this for the next release, but I haven't tackled it yet. I'll drop a line when I have some code you can test.

**Comment by saghul** [comment]
Can you try this branch? https://github.com/saghul/pyuv/tree/win-python3.5

**Comment by brcolow** [comment]
@saghul It works!

**Comment by saghul** [comment]
Awesome! I just released version 1.3.0, including binary Wheels, so you can avoid compiling it now. Cheers!
